netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -107,13 +109,15 @@ public static ProtocolNegotiator serverPlaintext() {
   /**
    * Create a server TLS handler for HTTP/2 capable of using ALPN/NPN.
    */
-  public static ProtocolNegotiator serverTls(final SslContext sslContext) {
+  public static ProtocolNegotiator serverTls(final SslContext sslContext,
+      final ObjectPool<? extends Executor> executorPool) {
     Preconditions.checkNotNull(sslContext, "sslContext");
+
```

Comment:
We don't want to create an `executor` every connection, so we should have the protocol negotiator keep a reference alive. 

The `executor` should be pulled out of the `executorPool` before returning and returned to the pool in `close()`. But it actually looks like server-side never calls close(). @jiangtaoli2016, is it possible your [alts protocol negotiator](https://github.com/grpc/grpc-java/blob/2de5b4e254a09f69555251151c2a8b950310bc9b/alts/src/main/java/io/grpc/alts/internal/AltsProtocolNegotiator.java#L116) hasn't been cleaning up the channel? On server-side I'd sort of expect close() to be called while some handlers may still be functioning, so that means you would continue passing the pool to the handler and let it _also_ get the executor from the pool.

Ditto for client-side, although on client-side we do guarantee that no connections exist when the negotiator is closed, so it is safe to pass Executor directly to the handler and only deal with the pool in the negotiator.

Reply:
IIUC, regardless of how alts protocol negotiator behaves, the current solution is:
1. on client side, we have the protocol negotiator with a reference to a `Executor` and clean it up when calling `close()`
2. on server side, since `close()` is not guaranteed to be called every time, we have to keep the current approach, which is to  have a reference to a `ExecutorPool` and create an `Executor` every connection

Reply:
You should rely on close(). If close is not functioning that is a bug that should be fixed, not worked around. You should not _just_ get a reference for each connection on server-side. That can lead to _horrible_ behavior with frequent destruction/creation of threads.

Reply:
#6508 fixes the NettyServer behavior to close the negotiator.